src/components/DynamicFormContainer/DynamicForm/DynamicForm.tsx

```diff
@@ -39,9 +39,36 @@ export const DynamicFormRaw: FunctionComponent<DynamicFormProps> = ({
   attachmentAcceptedFormat = "",
 }) => {
   const { templateIndex, data, ownership } = form;
-  const { newPopulatedForm } = useFormsContext();
+  const { newPopulatedForm, currentForm, setCurrentFileName } = useFormsContext();
   const isTransferableRecord = type === "TRANSFERABLE_RECORD";
 
+  const [newFileName, setNewFileName] = useState(currentForm?.fileName);
+
+  const mounted = useRef(false);
+  useEffect(() => {
+    if (!mounted.current) {
+      mounted.current = true;
+    } else {
+      setNewFileName(currentForm?.fileName);
+    }
+  }, [currentForm?.fileName]);
+
+  const debouncedChange = useMemo(
+    () =>
+      debounce((val) => {
+        setCurrentFileName(val);
+      }, 400),
+    [setCurrentFileName]
+  );
+
```

Comment:
Since u used denounce here, should U have a cancel denounce on unmount or component?

Reply:
no need to remove debounce, its not a listener

Reply:
i understand its not a listener, but i think its ok to not clean up this on unmount.